backend/src/v5/models/modelSettings.js

```diff
@@ -34,6 +38,38 @@ const getModelByQuery = async (ts, query, projection) => {
 	return res;
 };
 
+Models.addModel = async (ts, data) => {
+	const res = await insertOneModel(ts, { _id: generateUUIDString(), ...data });
+	return res.insertedId;
+};
+
+Models.deleteModel = async (ts, model) => {
+	const res = await deleteOneModel(ts, { _id: model });
+
+	if (res.deletedCount === 0) {
+		throw templates.modelNotFound;
+	}
+};
+
+Models.isSubModel = async (ts, model) => {
+	const subModelCount = await countModels(ts, { 'subModels.model': model });
+
+	return subModelCount > 0;
+};
+
+Models.removeModelCollections = async (ts, model) => {
+	const collections = await db.listCollections(ts);
+	const promises = [];
+
+	collections.forEach((collection) => {
+		if (collection.name.startsWith(`${model}.`)) {
```

Comment:
I think this logic should go up to the processor - this has nothing to do with modelSettings.